site/content/architecture/m3aggregator/flushing.md

```diff
@@ -0,0 +1,58 @@
+---
+title: "Flushing"
+weight: 2
+---
+
+## Overview
+
+Flushing is the process by which m3aggregator instances output the aggregated time series data 
+(by using [m3msg](https://github.com/m3db/m3/tree/master/src/msg#readme) protocol).
+There are two targets to which the data is flushed:
+1. Persistence: the aggregated data is flushed to `m3coordinator` which then 
+   persists it in `m3db`.
+1. Forwarding: intermediate aggregation data is flushed to other nodes of `m3aggregator`
+   (in a multi-node setup) for further processing. This is necessary for rollup rule 
+   processing - eg. when a label is dropped, the resulting time series has a new id
+   which potentially maps to a different shard (which may be owned by a different node).
+
+## Flushing coordination between leaders and followers 
```

Comment:
At the bottom of "Flushing coordination between leaders and followers" do you want to also perhaps briefly mention that if/when a follower is promoted to a leader that it will initially perform a flush any time windows that it has buffered but has not been flushed yet (according to the flushed times in etcd).

Reply:
👍  https://github.com/m3db/m3/pull/3578/commits/466dc9cc42fe6a32c36850d181e701044c81e24b